### PR TITLE
Fix: Pipedrive multiselect field sync 

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/251
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5163
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This bugfix/feature has been sponsored by @Webmecanik

When you try to sync. a select type field, the fields are mapped but not synced.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create new Custom multiselect field in Pipedrive with three optiosn (first, second, third)
2. Create new Custom multiselect field in Mautic with the same options = labels and values should be same as label at Pipedrive
3. Setup Pipedrive plugin and match Custom multiselect field from  Pipedrive to Mautic
4.  Create form with Multiselect field and match to Contact same contact field
5: Add Push contact to integration  action
6. Send form and see this field cannot be added to Pipedrive

#### Steps to test the bug:
1. Repeat steps
2. You should see update field in Pipedrive

image

### CODE REVIEW

**Comment by npracht** [comment]
Tested 👍

**Comment by npracht** [comment]
Hello @kuzmany !

I also just tested in the other way **Pipedrive --> mautic** for select and multiselect fields and it is not working properly.

## Steps to reproduce
1. Create a contact in PD with a select custom field (here `custom1` and `custom2`)
2. Sync the contact in mautic (map the custom field to a select mautic field)

Notice that stored in mautic are values without labels.
#### Pipedrive
image 2

#### Mautic result
image 1
Even if the custom contact field has been set with correct labels/values.

**Comment by kuzmany** [comment]
@npracht 
updated. 
Pipedrive send to webhook just Ids of multiselect options. 
This update try find label of each option and try syn to multiselect custom field in Mautic.
Please retest

**Comment by escopecz** [comment]
@kuzmany could you please fix the CS? https://travis-ci.org/mautic/mautic/jobs/332060106#L718

**Comment by javjim-mautic** [comment]
it worked for me but its also creating 2 contacts when its pipedrive ->mautic can you verify that @kuzmany

**Comment by escopecz** [comment]
I'm stuck on step 3. The multiselect field I created in Pipedrive does not show up in the mapping form in the Pipedrive plugin configuration. Are the fields cached somehow?

**Comment by kuzmany** [comment]
@javjim-mautic Hope this should fix it https://github.com/mautic/mautic/pull/5401 That PR should rework sync between Pipedrive <> Mautic and I think this issue was resolved there

**Comment by kuzmany** [comment]
Label: WIP

**Comment by kuzmany** [comment]
@escopecz How did you create Pipedrive custom field? 
I've just test it and works. Fields are not cached
Did you create custom field under People tab?

image

image

**Comment by kuzmany** [comment]
Label: Needs Documentation

**Comment by kuzmany** [comment]
Label: WIP

**Comment by kuzmany** [comment]
Label: needs documentation

**Comment by kuzmany** [comment]
Label: Pending Feedback

**Comment by kuzmany** [comment]
Label: Ready to test

**Comment by kuzmany** [comment]
Label: Needs Documentation

**Comment by escopecz** [comment]
Aha, that was it! I created it under **Deals** tab. Didn't notice those tabs at all. Will retest.

File: plugins/MauticCrmBundle/Integration/Pipedrive/AbstractPipedrive.php [hunk 1]
```diff
@@ -38,8 +38,32 @@ public function __construct(EntityManager $em)
...
     }
 
-    protected function convertPipedriveData(array $data = [])
+    /**
+     * @param array $data
+     * @param array $objectFields
+     *
+     * @return array
+     */
+    protected function convertPipedriveData(array $data = [], array $objectFields = [])
```
Comment (escopecz) [inline plugins/MauticCrmBundle/Integration/Pipedrive/AbstractPipedrive.php]: This method would deserve a unit test. @kuzmany could you please add it?

**Comment by escopecz** [comment]
Works as described 👍  Just the unit test would be really helpful to ensure the newly added data modifications work correctly and that it will stay that way with future modifications.

**Comment by kuzmany** [comment]
@escopecz 
Please, How did you test phpunit tests? 
Because when I run in past, Pipedrive tests returns some issue. 
But I will add it, no prob.

**Comment by escopecz** [comment]
You are right. There are some notices in the unit tests for PD: https://travis-ci.org/mautic/mautic/jobs/335114010#L646 . Fix those too if you can please.

To run just Pipedrive tests execute:
`bin/phpunit --bootstrap vendor/autoload.php --configuration app/phpunit.xml.dist --filter PipedriveTest`

Warning: Better backup your database before you run any tests. Some of the devs lost their data because of test database misconfiguration.

**Comment by javjim-mautic** [comment]
Tested and works properly

**Pull request merged**